File: backend/src/main/java/backend/mulkkam/cup/dto/response/CupResponse.java
package backend.mulkkam.cup.dto.response;

import backend.mulkkam.cup.domain.Cup;
import backend.mulkkam.cup.domain.IntakeType;
import io.swagger.v3.oas.annotations.media.Schema;

public record CupResponse(
        @Schema(description = "컵 식별자", example = "1")
        Long id,
        @Schema(description = "컵 이름", example = "스타벅스 텀블러")
        String cupNickname,
        @Schema(description = "컵 용량(ml)", example = "500")
        Integer cupAmount,
        @Schema(description = "컵 우선순위", example = "3")
        Integer cupRank,
        @Schema(description = "음료 종류", implementation = IntakeType.class)
        IntakeType intakeType,
        @Schema(description = "이모지", example = "☕️")
        String emoji
) {

    public CupResponse(Cup cup) {
        this(
                cup.getId(),
                cup.getNickname().value(),
                cup.getCupAmount().value(),
                cup.getCupRank().value(),
                cup.getIntakeType(),
                cup.getEmoji()
        );
    }
}

File: android/app/src/main/java/com/mulkkam/data/remote/model/response/CupResponse.kt
package com.mulkkam.data.remote.model.response

import com.mulkkam.domain.model.Cup
import com.mulkkam.domain.model.IntakeType
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class CupResponse(
    @SerialName("id")
    val id: Int,
    @SerialName("cupNickname")
    val cupNickname: String,
    @SerialName("cupAmount")
    val cupAmount: Int,
    @SerialName("cupRank")
    val cupRank: Int,
    @SerialName("intakeType")
    val intakeType: String,
    @SerialName("emoji")
    val emoji: String,
)

fun CupResponse.toDomain() =
    Cup(
        id = id,
        nickname = cupNickname,
        amount = cupAmount,
        rank = cupRank,
        intakeType = IntakeType.from(intakeType),
        emoji = emoji,
    )
